docker/docker_client.go

```diff
@@ -548,34 +552,36 @@ func (c *dockerClient) setupRequestAuth(req *http.Request, extraScope *authScope
 	return nil
 }
 
-func (c *dockerClient) getBearerToken(ctx context.Context, challenge challenge, scopes []authScope) (*bearerToken, error) {
+func (c *dockerClient) getBearerTokenBasic(ctx context.Context, challenge challenge, scopes []authScope) (*bearerToken, error) {
 	realm, ok := challenge.Parameters["realm"]
 	if !ok {
 		return nil, errors.Errorf("missing realm in bearer auth challenge")
 	}
 
-	authReq, err := http.NewRequest("GET", realm, nil)
+	authReq, err := http.NewRequest(http.MethodGet, realm, nil)
 	if err != nil {
 		return nil, err
 	}
+
 	authReq = authReq.WithContext(ctx)
-	getParams := authReq.URL.Query()
-	if c.username != "" {
-		getParams.Add("account", c.username)
```

Comment:
Why is it OK to drop this code?

Reply:
Base on the document "account" is not one of the parameters specified. However, "offline" and "client_id" are two params that are missing from the current implementation, but seems to be OK if not set. I didn't find a reason why this was originally added, so I dropped it here.

Ref: https://docs.docker.com/registry/spec/auth/token/

Reply:
Git history points at https://github.com/containers/image/pull/480 .

Specs lie, please don’t remove things unrelated to the purpose of the PR.